File: 06-TREES/PHP/insert-into-a-binary-search-tree.php
<?php

# https://leetcode.com/problems/insert-into-a-binary-search-tree/
class Solution
{
    public function insertIntoBST(?TreeNode $root, int $val): TreeNode
    {
        if (!$root)
            return new TreeNode($val);

        if ($val > $root->val)
            $root->right = $this->insertIntoBST($root->right, $val);
        elseif ($val < $root->val)
            $root->left = $this->insertIntoBST($root->left, $val);

        return $root;
    }
}

File: 06-TREES/Go/insert-into-a-binary-search-tree.go
package main

// https://leetcode.com/problems/insert-into-a-binary-search-tree/

import "../../libs/Go/tree"

func insertIntoBST(root *tree.Node, val int) *tree.Node {
	if root == nil {
		return &tree.Node{Val: val}
	}

	if root.Val > val {
		root.Left = insertIntoBST(root.Left, val)
	} else if root.Val < val {
		root.Right = insertIntoBST(root.Right, val)
	}

	return root
}
